src/build.cc

```diff
@@ -679,14 +679,33 @@ bool Builder::StartEdge(Edge* edge, string* err) {
 
   status_->BuildEdgeStarted(edge, start_time_millis);
 
-  // Create directories necessary for outputs.
+  TimeStamp build_start = -1;
+
+  // Create directories necessary for outputs and remember the current
+  // filesystem mtime to record later
   // XXX: this will block; do we care?
   for (vector<Node*>::iterator o = edge->outputs_.begin();
        o != edge->outputs_.end(); ++o) {
     if (!disk_interface_->MakeDirs((*o)->path()))
       return false;
+    if (build_start == -1) {
+      string tmp_path = DirName((*o)->PathDecanonicalized(), true) + ".ninja_tmp";
+      if (!disk_interface_->WriteFile(tmp_path, "")) {
+        err->assign("could not create temporary file to determine command "
+                    "start time of '" + edge->EvaluateCommand() + "'");
+        return false;
+      }
+
+      build_start = disk_interface_->Stat(tmp_path, err);
+      disk_interface_->RemoveFile(tmp_path);
+
+      if (build_start == -1)
+        return false;
```

Comment:
ninja shouldn't fail if it can't create the file, but rather behave as before.

Reply:
done